### PR TITLE
[Bug](jdbc) Fix memory leak for JDBC datasource

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

1. Fix memory leak.
2. Use `hikari` to manage JDBC connections.
3. Do some refactor, using executeQuery interface for Select statement to achieve better performance

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 1]
```diff
@@ -41,22 +42,26 @@ const char* JDBC_EXECUTOR_TRANSACTION_SIGNATURE = "()V";
...
+        RETURN_IF_ERROR(abort_trans());
     }
     JNIEnv* env;
-    Status status;
-    RETURN_IF_STATUS_ERROR(status, JniUtil::GetJNIEnv(&env));
+    RETURN_IF_ERROR(JniUtil::GetJNIEnv(&env));
+    env->DeleteGlobalRef(_executor_clazz);
+    env->DeleteGlobalRef(_executor_list_clazz);
+    env->DeleteGlobalRef(_executor_object_clazz);
+    env->DeleteGlobalRef(_executor_string_clazz);
```
Comment (zhangstar333) [inline be/src/vec/exec/vjdbc_connector.cpp:57]: could add those variables:  jclass _executor_##RETURN_TYPE##_clazz;

File: be/src/vec/exec/vjdbc_connector.cpp [hunk 2]
```diff
@@ -121,7 +128,7 @@ Status JdbcConnector::query() {
     RETURN_IF_ERROR(JniUtil::GetJNIEnv(&env));
     jstring query_sql = env->NewStringUTF(_sql_str.c_str());
     jint colunm_count = env->CallNonvirtualIntMethod(_executor_obj, _executor_clazz,
-                                                     _executor_query_id, query_sql);
+                                                     _executor_read_id, query_sql);
```
Comment (zhangstar333) [inline be/src/vec/exec/vjdbc_connector.cpp:131]: I see register read function is `int read() `?

File: be/src/util/jni-util.cpp [hunk 3]
```diff
@@ -39,13 +39,17 @@ void FindOrCreateJavaVM() {
     if (rv == 0) {
         JNIEnv* env;
         JavaVMInitArgs vm_args;
-        JavaVMOption options[1];
+        JavaVMOption options[2];
         char* str = getenv("DORIS_JNI_CLASSPATH_PARAMETER");
         options[0].optionString = str;
+        // Set max heap size to 4G.
+        std::string heap_size("-Xmx4096M");
```
Comment (morningman) [inline be/src/util/jni-util.cpp:46]: This should be a `config`

File: fe/java-udf/pom.xml [hunk 4]
```diff
@@ -70,6 +70,11 @@ under the License.
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.zaxxer</groupId>
+            <artifactId>HikariCP</artifactId>
+            <version>${hikaricp.version}</version>
```
Comment (morningman) [inline fe/java-udf/pom.xml:76]: Add this to `dist/LICENSE-dist.txt`

**Review by morningman — DISMISSED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**